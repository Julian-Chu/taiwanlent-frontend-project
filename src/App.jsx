import React, { Component } from 'react';
import SidePanel from './components/SidePanelComponent';
import Header from './components/HeaderComponent';
import Home from './components/home/Home';
import Talents from './components/talents/Talents';
import Footer from './components/Footer';
import './styles/App.css';
import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
import Login from './components/login/Login';
// import User from './components/User';
import BusinessUserSignUp from './components/businessUser/BusinessUserSignUp';
import Logout from './components/Logout';
import BusinessUserRegister_old from './components/businessUser/BusinessUserRegister_old';
import BusinessUserRegister from './components/businessUser/BusinessUserRegister';
import BusinessUserProfile from './components/businessUser/BusinessUserProfile';
import WelcomeNewUser from './components/WelcomeNewUser';
import PersonalUserRegister from './components/personalUser/PersonalUserRegister';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = {
            sidePanelIsOpened: false,
            headerIsTransparent: true,
        }
        this.handleScroll = this.handleScroll.bind(this);
    }

    toggleSidePanelOpen() {
        this.setState({
            sidePanelIsOpened: !this.state.sidePanelIsOpened
        })
    }

    subscribeHeaderTransparentEvent() {
        this.handleScrollTemp = this.debounce(this.handleScroll, 100)
        window.addEventListener('scroll', this.handleScrollTemp);
    }

    unsubscribeHeaderTransparentEvent() {
        if (this.handleScrollTemp !== null)
            window.removeEventListener('scroll', this.handleScrollTemp);
    }

    handleScroll() {
        let rectHeader = document.querySelector('#header').getBoundingClientRect();
        const contentNode = document.querySelector('#content');
        let rectContent = contentNode? contentNode.getBoundingClientRect(): {top:0};

        if ((rectHeader.top + rectContent.top) < 0) {
            if (this.state.headerIsTransparent === true)
                this.setState({ headerIsTransparent: false });
        } else {
            if (this.state.headerIsTransparent === false)
                this.setState({ headerIsTransparent: true });
        }

    }

    debounce(func, wait, immediate) {
        var timeout;
        return function () {
            var context = this, args = arguments;
            var later = function () {
                timeout = null;
                if (!immediate) func.apply(context, args);
            };
            var callNow = immediate && !timeout;
            clearTimeout(timeout);
            timeout = setTimeout(later, wait);
            if (callNow) func.apply(context, args);
        };
    }

    setHeaderTransparent(bool) {
        if (this.state.headerIsTransparent !== bool)
            this.setState({
                headerIsTransparent: bool
            });
    }

    render() {
        return (

            <div className={this.state.sidePanelIsOpened ? 'side-panel-open' : ''}>
                <div className="body-overlay" onClick={() => this.toggleSidePanelOpen()}></div>
                <SidePanel></SidePanel>
                <div id="container" className="clear-fix">
                    <BrowserRouter>
                        <div>
                            <Header toggleSidePanelOpen={() => this.toggleSidePanelOpen()}
                                headerIsTransparent={this.state.headerIsTransparent}
                            ></Header>
                            <Switch>
                                    <Route path="/home"
                                        exact component={
                                            (props) => <Home
                                                {...props}
                                                subscribeTransparentEvent={() => this.subscribeHeaderTransparentEvent()}
                                                unsubscribeTransparentEvent={() => this.unsubscribeHeaderTransparentEvent()}
                                            />} />
                                            
                                    <Route path="/talents" extact component={(props) => <Talents
                                        {...props}
                                        setHeaderNontransparent={() => this.setHeaderTransparent(false)}
                                    />} />
                                    <Route path="/login" extact component={(props) => <Login
                                        {...props}
                                        setHeaderNontransparent={() => this.setHeaderTransparent(false)}
                                    />} />
                                    {/* <Route path="/user" extact component={(props) => <User
                                        setHeaderNontransparent={() => this.setHeaderTransparent(false)}
                                        {...props}
                                    />} /> */}

                                    <Route path="/businessUserData" extact component={(props)=><BusinessUserProfile {...props}/>}></Route>
                                    <Route path="/businessUserSignUp" extact component={(props) => <BusinessUserSignUp
                                        setHeaderNontransparent={() => this.setHeaderTransparent(false)}
                                        {...props}
                                    />} />
                                    <Route path="/personalUserRegister" extact component={(props) => <PersonalUserRegister {...props}/>}></Route>
                                    <Route path="/WelcomeNewUser" exact component={(props)=><WelcomeNewUser {...props} />}></Route>
                                    {/* For test */}
                                    <Route path="/businessUserRegister" extact component={(props)=> <BusinessUserRegister {...props}/>  }/>
                                    <Route path="/logout" extact component={props => <Logout
                                        setHeaderNontransparent={() => this.setHeaderTransparent(false)}
                                        {...props}
                                    />} />
                                    <Redirect to="/home" />
                            </Switch>
                        </div>
                    </BrowserRouter>
                </div>
                <Footer></Footer>
            </div>
        );
    }
}

export default App;
