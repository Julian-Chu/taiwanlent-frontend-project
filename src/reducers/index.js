import { combineReducers} from 'redux';
import TalentsReducer from './reducer_talents';
import FilterReducer from './reducer_filters';

const rootReducer = combineReducers({
  talents: TalentsReducer,
  filter: FilterReducer
});

export default rootReducer;