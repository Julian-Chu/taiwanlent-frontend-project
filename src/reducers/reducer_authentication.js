import {
  CHANGE_AUTH,
  LOGOUT
} from '../actions/types';

export default function (state = false, action) {
  switch (action.type) {
    case CHANGE_AUTH:
      console.log('CHANGE_AUTH');
      return action.payload;
    case LOGOUT:
      console.log('Logout');
      return false;
    default:
      console.log('requiredAuth state:', state);
      return state;
  }
}