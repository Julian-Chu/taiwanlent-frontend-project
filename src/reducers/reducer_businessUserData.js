import {BUSINESS_USER_DATA} from '../actions/types';

export default function(state=null, action){
  switch(action.type){
    case BUSINESS_USER_DATA:
      return action.payload;
    default:
      return state;
  }

}