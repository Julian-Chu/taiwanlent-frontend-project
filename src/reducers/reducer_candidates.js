import {
    AddCandidateToList
} from '../actions/index';
import {
    RemoveCandidateFromList
} from '../actions/index';
import {
    RemoveAllCandidates
} from '../actions/index';



export default function (state = [], action) {
    let newState = state.slice();
    switch (action.type) {
        case AddCandidateToList:
            if (state.indexOf(action.payload) === -1) // check existing id in list
                newState = [action.payload, ...state];
            console.log('Add Candidate, new list:', newState);
            break;
        case RemoveCandidateFromList:
            newState = state.slice();
            newState.splice(newState.indexOf(action.payload), 1);
            console.log('Remove Candidate, new list:', newState);
            break;
        case RemoveAllCandidates:
            newState = [];
            break;
        default:
            break;
        }
        return newState;
}