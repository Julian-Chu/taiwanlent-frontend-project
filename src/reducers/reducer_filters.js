import {FILTERS} from '../actions/index';

export default function(state=[], action){
  console.log(action.payload);
  switch(action.type){    
    case FILTERS:
      return [...action.payload];
    default:
      return state;
  }
}