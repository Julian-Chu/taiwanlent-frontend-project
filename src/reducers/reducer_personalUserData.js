import {
  PERSONAL_USER_DATA
} from '../actions/types';

export default function (state = {}, action) {
  switch (action.type) {
    case PERSONAL_USER_DATA:
      console.log("personal user data:", action.payload);
      return action.payload;
    default:
      return state;
  }
}