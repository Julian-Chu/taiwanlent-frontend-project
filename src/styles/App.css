.side-panel-open .body-overlay {
    opacity: 1;
    z-index: 298;
    pointer-events: auto;
}

.body-overlay {
    opacity: 0;
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.6);
    z-index: -1;
    pointer-events: none;
    -webkit-backface-visibility: hidden;
    -webkit-transition: opacity .3s ease;
    -o-transition: opacity .3s ease;
    transition: opacity .3s ease;
}

.side-panel-open:not(.device-xxs):not(.device-xs):not(.device-sm) .body-overlay:hover {
    cursor: url('../images/close.png') 15 15, default;
}

#side-panel {
	position: fixed;
	top: 0;
	right: -300px;
	z-index: 399;
	width: 300px;
	height: 100%;
	background-color: #F5F5F5;
	overflow: hidden;
	-webkit-backface-visibility: hidden;
}

.side-panel-open #side-panel {
    right: 0;
}



/* .side-push-panel.stretched .side-panel-open #wrapper,
.side-push-panel.stretched .side-panel-open #header.sticky-header .container, 
{
    right: 300px;
} */

.side-push-panel.stretched .side-panel-open .slider-parallax-inner,
.side-push-panel.stretched .side-panel-open #content,
.side-push-panel.stretched .side-panel-open header,
.side-push-panel.stretched .side-panel-open #container,
.side-push-panel.stretched .side-panel-open #footer  {
    left: -300px;
}

/* #side-panel-trigger {
    display: block;
    cursor: pointer;
    z-index: 11;
    margin-right: 5px;
} */

/* #side-panel-trigger-close a {
    display: none;
} */

/* #side-panel,
.side-push-panel.stretched #wrapper,
.side-push-panel.stretched #header .container {
    -webkit-transition: right .4s ease;
    -o-transition: right .4s ease;
    transition: right .4s ease;
} */