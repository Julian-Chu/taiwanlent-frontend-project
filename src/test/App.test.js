import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import {shallow, mount} from 'enzyme';
import {Provider} from 'react-redux';
import { createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import reducers from '../reducers';

describe('App', ()=>{
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={createStoreWithMiddleware(reducers)}><App/></Provider>, div);
});

it('renders correctly',()=>{
  expect(app).toMatchSnapshot();
})

console.log('app:',app);
it('Class body-overlay exists',()=>{
  expect(app.find('.body-overlay').exists()).toBe(true);
})

it('SidePanel exists',()=>{
  expect(app.find('SidePanel').exists()).toBe(true);
})

it('Header exists',()=>{
  expect(app.find('Header').exists()).toBe(true);
})

it('Footer exists',()=>{
  expect(app.find('Footer').exists()).toBe(true);
})

});
