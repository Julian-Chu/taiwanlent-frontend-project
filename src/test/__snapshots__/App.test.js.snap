// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="body-overlay"
          onClick={[Function]}
/>,
        <Connect(SidePanel) />,
        <div
          className="clear-fix"
          id="container"
>
          <BrowserRouter>
                    <div>
                              <Connect(Header)
                                        headerIsTransparent={true}
                                        toggleSidePanelOpen={[Function]}
                              />
                              <Switch>
                                        <Route
                                                  component={[Function]}
                                                  exact={true}
                                                  path="/home"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  extact={true}
                                                  path="/talents"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  extact={true}
                                                  path="/login"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  extact={true}
                                                  path="/user"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  extact={true}
                                                  path="/businessUserInitRegister"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  extact={true}
                                                  path="/businessUserRegister"
                                        />
                                        <Route
                                                  component={[Function]}
                                                  extact={true}
                                                  path="/logout"
                                        />
                                        <Redirect
                                                  push={false}
                                                  to="/home"
                                        />
                              </Switch>
                    </div>
          </BrowserRouter>
</div>,
        <Footer />,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "body-overlay",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <BrowserRouter>
            <div>
                        <Connect(Header)
                                    headerIsTransparent={true}
                                    toggleSidePanelOpen={[Function]}
                        />
                        <Switch>
                                    <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/home"
                                    />
                                    <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/talents"
                                    />
                                    <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/login"
                                    />
                                    <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/user"
                                    />
                                    <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/businessUserInitRegister"
                                    />
                                    <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/businessUserRegister"
                                    />
                                    <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/logout"
                                    />
                                    <Redirect
                                                push={false}
                                                to="/home"
                                    />
                        </Switch>
            </div>
</BrowserRouter>,
          "className": "clear-fix",
          "id": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              <Connect(Header)
                            headerIsTransparent={true}
                            toggleSidePanelOpen={[Function]}
              />
              <Switch>
                            <Route
                                          component={[Function]}
                                          exact={true}
                                          path="/home"
                            />
                            <Route
                                          component={[Function]}
                                          extact={true}
                                          path="/talents"
                            />
                            <Route
                                          component={[Function]}
                                          extact={true}
                                          path="/login"
                            />
                            <Route
                                          component={[Function]}
                                          extact={true}
                                          path="/user"
                            />
                            <Route
                                          component={[Function]}
                                          extact={true}
                                          path="/businessUserInitRegister"
                            />
                            <Route
                                          component={[Function]}
                                          extact={true}
                                          path="/businessUserRegister"
                            />
                            <Route
                                          component={[Function]}
                                          extact={true}
                                          path="/logout"
                            />
                            <Redirect
                                          push={false}
                                          to="/home"
                            />
              </Switch>
</div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(Header)
                  headerIsTransparent={true}
                  toggleSidePanelOpen={[Function]}
/>,
                <Switch>
                  <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/home"
                  />
                  <Route
                                    component={[Function]}
                                    extact={true}
                                    path="/talents"
                  />
                  <Route
                                    component={[Function]}
                                    extact={true}
                                    path="/login"
                  />
                  <Route
                                    component={[Function]}
                                    extact={true}
                                    path="/user"
                  />
                  <Route
                                    component={[Function]}
                                    extact={true}
                                    path="/businessUserInitRegister"
                  />
                  <Route
                                    component={[Function]}
                                    extact={true}
                                    path="/businessUserRegister"
                  />
                  <Route
                                    component={[Function]}
                                    extact={true}
                                    path="/logout"
                  />
                  <Redirect
                                    push={false}
                                    to="/home"
                  />
</Switch>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "headerIsTransparent": true,
                  "toggleSidePanelOpen": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Route
                      component={[Function]}
                      exact={true}
                      path="/home"
/>,
                    <Route
                      component={[Function]}
                      extact={true}
                      path="/talents"
/>,
                    <Route
                      component={[Function]}
                      extact={true}
                      path="/login"
/>,
                    <Route
                      component={[Function]}
                      extact={true}
                      path="/user"
/>,
                    <Route
                      component={[Function]}
                      extact={true}
                      path="/businessUserInitRegister"
/>,
                    <Route
                      component={[Function]}
                      extact={true}
                      path="/businessUserRegister"
/>,
                    <Route
                      component={[Function]}
                      extact={true}
                      path="/logout"
/>,
                    <Redirect
                      push={false}
                      to="/home"
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "exact": true,
                      "path": "/home",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "extact": true,
                      "path": "/talents",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "extact": true,
                      "path": "/login",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "extact": true,
                      "path": "/user",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "extact": true,
                      "path": "/businessUserInitRegister",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "extact": true,
                      "path": "/businessUserRegister",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "component": [Function],
                      "extact": true,
                      "path": "/logout",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "push": false,
                      "to": "/home",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="body-overlay"
            onClick={[Function]}
/>,
          <Connect(SidePanel) />,
          <div
            className="clear-fix"
            id="container"
>
            <BrowserRouter>
                        <div>
                                    <Connect(Header)
                                                headerIsTransparent={true}
                                                toggleSidePanelOpen={[Function]}
                                    />
                                    <Switch>
                                                <Route
                                                            component={[Function]}
                                                            exact={true}
                                                            path="/home"
                                                />
                                                <Route
                                                            component={[Function]}
                                                            extact={true}
                                                            path="/talents"
                                                />
                                                <Route
                                                            component={[Function]}
                                                            extact={true}
                                                            path="/login"
                                                />
                                                <Route
                                                            component={[Function]}
                                                            extact={true}
                                                            path="/user"
                                                />
                                                <Route
                                                            component={[Function]}
                                                            extact={true}
                                                            path="/businessUserInitRegister"
                                                />
                                                <Route
                                                            component={[Function]}
                                                            extact={true}
                                                            path="/businessUserRegister"
                                                />
                                                <Route
                                                            component={[Function]}
                                                            extact={true}
                                                            path="/logout"
                                                />
                                                <Redirect
                                                            push={false}
                                                            to="/home"
                                                />
                                    </Switch>
                        </div>
            </BrowserRouter>
</div>,
          <Footer />,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "body-overlay",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <BrowserRouter>
              <div>
                            <Connect(Header)
                                          headerIsTransparent={true}
                                          toggleSidePanelOpen={[Function]}
                            />
                            <Switch>
                                          <Route
                                                        component={[Function]}
                                                        exact={true}
                                                        path="/home"
                                          />
                                          <Route
                                                        component={[Function]}
                                                        extact={true}
                                                        path="/talents"
                                          />
                                          <Route
                                                        component={[Function]}
                                                        extact={true}
                                                        path="/login"
                                          />
                                          <Route
                                                        component={[Function]}
                                                        extact={true}
                                                        path="/user"
                                          />
                                          <Route
                                                        component={[Function]}
                                                        extact={true}
                                                        path="/businessUserInitRegister"
                                          />
                                          <Route
                                                        component={[Function]}
                                                        extact={true}
                                                        path="/businessUserRegister"
                                          />
                                          <Route
                                                        component={[Function]}
                                                        extact={true}
                                                        path="/logout"
                                          />
                                          <Redirect
                                                        push={false}
                                                        to="/home"
                                          />
                            </Switch>
              </div>
</BrowserRouter>,
            "className": "clear-fix",
            "id": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div>
                <Connect(Header)
                                headerIsTransparent={true}
                                toggleSidePanelOpen={[Function]}
                />
                <Switch>
                                <Route
                                                component={[Function]}
                                                exact={true}
                                                path="/home"
                                />
                                <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/talents"
                                />
                                <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/login"
                                />
                                <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/user"
                                />
                                <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/businessUserInitRegister"
                                />
                                <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/businessUserRegister"
                                />
                                <Route
                                                component={[Function]}
                                                extact={true}
                                                path="/logout"
                                />
                                <Redirect
                                                push={false}
                                                to="/home"
                                />
                </Switch>
</div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(Header)
                    headerIsTransparent={true}
                    toggleSidePanelOpen={[Function]}
/>,
                  <Switch>
                    <Route
                                        component={[Function]}
                                        exact={true}
                                        path="/home"
                    />
                    <Route
                                        component={[Function]}
                                        extact={true}
                                        path="/talents"
                    />
                    <Route
                                        component={[Function]}
                                        extact={true}
                                        path="/login"
                    />
                    <Route
                                        component={[Function]}
                                        extact={true}
                                        path="/user"
                    />
                    <Route
                                        component={[Function]}
                                        extact={true}
                                        path="/businessUserInitRegister"
                    />
                    <Route
                                        component={[Function]}
                                        extact={true}
                                        path="/businessUserRegister"
                    />
                    <Route
                                        component={[Function]}
                                        extact={true}
                                        path="/logout"
                    />
                    <Redirect
                                        push={false}
                                        to="/home"
                    />
</Switch>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "headerIsTransparent": true,
                    "toggleSidePanelOpen": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <Route
                        component={[Function]}
                        exact={true}
                        path="/home"
/>,
                      <Route
                        component={[Function]}
                        extact={true}
                        path="/talents"
/>,
                      <Route
                        component={[Function]}
                        extact={true}
                        path="/login"
/>,
                      <Route
                        component={[Function]}
                        extact={true}
                        path="/user"
/>,
                      <Route
                        component={[Function]}
                        extact={true}
                        path="/businessUserInitRegister"
/>,
                      <Route
                        component={[Function]}
                        extact={true}
                        path="/businessUserRegister"
/>,
                      <Route
                        component={[Function]}
                        extact={true}
                        path="/logout"
/>,
                      <Redirect
                        push={false}
                        to="/home"
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "exact": true,
                        "path": "/home",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "extact": true,
                        "path": "/talents",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "extact": true,
                        "path": "/login",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "extact": true,
                        "path": "/user",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "extact": true,
                        "path": "/businessUserInitRegister",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "extact": true,
                        "path": "/businessUserRegister",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "component": [Function],
                        "extact": true,
                        "path": "/logout",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "push": false,
                        "to": "/home",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
